File: frontend/lib/api/risk.ts
import { apiClient } from './client'

export async function getRiskLimits() {
  const { data } = await apiClient.get('/risk/limits')
  return data
}

export async function patchRiskLimits(patch: Record<string, unknown>, confirmWidening = false) {
  const { data } = await apiClient.patch('/risk/limits', { ...patch, confirm_widening: confirmWidening })
  return data
}

export type RiskEvent = {
  time: string
  tier: string
  event: string
  trigger: string
  value: string
  action: string
  auto_recovered: boolean
}

export type RiskEventsResponse = {
  data: RiskEvent[]
  total: number
  days: number
}

export async function getRiskEvents(days = 30): Promise<RiskEventsResponse> {
  const { data } = await apiClient.get<RiskEventsResponse>('/risk/events', { params: { days } })
  return data
}


export type RiskThresholds = {
  daily_dd_halt_pct: string
  weekly_dd_halt_pct: string
  min_margin_usage_pct: string
  max_exchange_concentration_pct: string
  max_symbol_concentration_pct: string
  max_binance_mmr_pct: string
}

export async function getRiskThresholds(): Promise<RiskThresholds> {
  const { data } = await apiClient.get<RiskThresholds>("/risk/thresholds")
  return data
}

File: backend/app/api/v1/schemas/risk.py
"""Risk 请求 / 响应 schema。"""
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class RiskLimitsOut(BaseModel):
    max_positions: int
    stop_loss_pct: str
    max_hold_hours: str
    min_apr_pct: str
    max_total_notional_usd: str
    scan_threshold_apr_pct: str = "0"   # 0 = 回退用 min_apr_pct


class RiskLimitsPatch(BaseModel):
    max_positions: int | None = None
    stop_loss_pct: str | None = None
    max_hold_hours: str | None = None
    min_apr_pct: str | None = None
    max_total_notional_usd: str | None = None
    scan_threshold_apr_pct: str | None = None
    confirm_widening: bool = False


class RiskEventOut(BaseModel):
    time: datetime
    tier: str
    event: str
    trigger: str
    value: str
    action: str
    auto_recovered: bool


class RiskEventsResponse(BaseModel):
    data: list[RiskEventOut]
    total: int
    days: int
